frontend/server/src/Controllers/Submission.php

```diff
@@ -22,9 +22,31 @@ public static function apiLatestSubmissions(\OmegaUp\Request $r) {
         $offset = is_null($r['offset']) ? 1 : intval($r['offset']);
         $rowCount = is_null($r['rowcount']) ? 100 : intval($r['rowcount']);
 
+        $identityId = null;
+        if (!is_null($r['username'])) {
+            \OmegaUp\Validators::validateValidUsername(
+                $r['username'],
+                'username'
+            );
+
+            $user = \OmegaUp\DAO\Users::FindByUsername($r['username']);
+            if (!is_null($user) && $user->is_private) {
+                throw new \OmegaUp\Exceptions\ForbiddenAccessException(
+                    'userInformationIsPrivate'
+                );
+            }
+
+            $identity = \OmegaUp\DAO\Identities::FindByUsername($r['username']);
+            if (is_null($identity)) {
+                throw new \OmegaUp\Exceptions\NotFoundException('userNotExist');
+            }
+            $identityId = $identity->identity_id;
```

Comment:
casi! nada más te faltó la parte de "si es la identidad principal del usuario".

```suggestion
            $identity = \OmegaUp\DAO\Identities::FindByUsername($r['username']);
            if (is_null($identity)) {
                throw new \OmegaUp\Exceptions\NotFoundException('userNotExist');
            }

            $user = \OmegaUp\DAO\Users::FindByUsername($r['username']);
            if (!is_null($user) && $user->main_identity_id == $identity->identity_id && $user->is_private) {
                // Only the user's main identity is private.
                throw new \OmegaUp\Exceptions\ForbiddenAccessException(
                    'userInformationIsPrivate'
                );
            }
```